core/shared/src/main/scala/org/http4s/Header.scala

```diff
@@ -170,20 +170,28 @@ object Header {
         h: H
     )(implicit H: Header[H, _]): Header.ToRaw with Primitive =
       new Header.ToRaw with Primitive {
-        val values = List(Header.Raw(H.name, H.value(h)))
+        val values = Header.Raw(H.name, H.value(h)) :: Nil
       }
 
     implicit def foldablesToRaw[F[_]: Foldable, H](
         h: F[H]
     )(implicit convert: H => ToRaw with Primitive): Header.ToRaw = new Header.ToRaw {
-      val values: List[Raw] = h.toList.foldMap(v => convert(v).values)
+      val values = h
+        .foldLeft(collection.mutable.ListBuffer.empty[Header.Raw]) { (buf, v) =>
+          buf ++= convert(v).values
+        }
```

Comment:
Why don't use a regular for-each loop here?

Reply:
Does `cats.Foldable` has `foreach`?

Reply:
Ahhh, I see now!